### ISSUE TITLE
Add Duo support or describe a Duo workaround

### ISSUE THREAD

**Issue by jfhamlin** [oktadeveloper/okta-aws-cli-assume-role#58]
I'm moving to Duo from SMS for MFA, but I'm blocked on lack of Duo support in the cli. What is preventing support for Duo? I'm happy to contribute if it just takes some programming effort. Alternatively, can you offer a workaround for AWS CLI/API access when using Duo?

**Comment by AlainODea** [comment]
Once there is a release based on #109, you will be to get this working. I have explicitly tested this to be fixed. If you want this prior to that point, you can build master using maven and try it.

**Comment by AlainODea** [comment]
@mraible this can be closed. It’s fixed by #109 and is in [v1.0.2](https://github.com/oktadeveloper/okta-aws-cli-assume-role) 😊

**Comment by AlainODea** [comment]
@jfhamlin please try v1.0.2 with OKTA_BROWSESR_AUTH=true in okta.properties.

**Comment by jfhamlin** [comment]
Thanks! I’ll give it a try

**Comment by jimsmith** [comment]
@AlainODea 
Hello,

Tried v.1.0.2 this doesn't work throw's an Exception:
```
java -classpath ~/.okta/okta-aws-cli-1.0.2.jar com.okta.tools.awscli $@
Exception in thread "main" java.lang.NoClassDefFoundError: javafx/application/Application
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at com.okta.tools.saml.OktaSaml.getSamlResponse(OktaSaml.java:39)
        at com.okta.tools.OktaAwsCliAssumeRole.run(OktaAwsCliAssumeRole.java:88)
        at com.okta.tools.awscli.main(awscli.java:33)
Caused by: java.lang.ClassNotFoundException: javafx.application.Application
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 15 more
[vagrant@centos7 ~]$ java -version
openjdk version "1.8.0_161"
OpenJDK Runtime Environment (build 1.8.0_161-b14)
OpenJDK 64-Bit Server VM (build 25.161-b14, mixed mode)
```

Java installed `sudo yum install java-1.8.0-openjdk-headless.x86_64`


v1.0.1 gives different error, which I can confirm the same username and password I do have access to AWS through Okta via web https://REDACTED.okta.com/

```
OKTA_ORG=REDACTED.okta.com
OKTA_AWS_APP_URL=https://REDACTED.okta.com/home/amazon_aws/REDACTED/REDACTED
OKTA_USERNAME=
OKTA_PASSWORD=
OKTA_AWS_ROLE_TO_ASSUME=
OKTA_PROFILE=
OKTA_STS_DURATION=43200
OKTA_BROWSER_AUTH=true
```

```
java -classpath ~/.okta/okta-aws-cli-1.0.1.jar com.okta.tools.awscli $@
Username: REDACTED
Password:
Exception in thread "main" java.lang.RuntimeException: You do not have access to AWS through Okta.
Please contact your administrator.
        at com.okta.tools.saml.OktaSaml.getSamlResponseForAws(OktaSaml.java:54)
        at com.okta.tools.saml.OktaSaml.getSamlResponse(OktaSaml.java:44)
        at com.okta.tools.OktaAwsCliAssumeRole.run(OktaAwsCliAssumeRole.java:88)
        at com.okta.tools.awscli.main(awscli.java:33)
```

**Comment by AlainODea** [comment]
@jimmarks the first error is because BROWSER_AUTH depends on Java FX and your JRE doesn’t have it.

Java FX may not be included in the openjdk8-jre package on many Linux distros. You may need the openjfx package instead. https://stackoverflow.com/a/19529820

You could also install the Oracle JDK and that should work.

I haven’t tried this on Linux.

The second error looks like app-level MFA is enabled. BROWSER_AUTH is a potential workaround and only became available in v1.0.2.

**Comment by AlainODea** [comment]
@jimsmith **OKTA_BROWSER_AUTH** needs a graphical desktop environment. There is no supported way to do Duo Push authentication with Okta without a graphical web browser.

You will not be able to use Duo Push as the factor auth in the setup you are describing.

I recommend that you remove **OKTA_BROWSER_AUTH** from **okta.properties** and use the Google Authenticator MFA option. You can scan the QR code Okta presents during enrollment with any TOTP-compatible virtual MFA device (the Duo Mobile iOS and Android apps included).
